File: ReclamationNour/reclamtion/src/main/java/com/reclamation/reclamtion/service/ReclamationService.java
package com.reclamation.reclamtion.service;

import java.util.List;

import com.reclamation.reclamtion.model.Reclamation;


public interface  ReclamationService {

    List<Reclamation> getAllReclamations();

  
    Reclamation getReclamationById(Long id);


    Reclamation createReclamation(Reclamation reclamation);

 
    Reclamation updateReclamation(Long id, Reclamation updatedReclamation);


    void deleteReclamation(Long id);


    
}

File: care4elder-backoffice-master/src/app/services/reclamation.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Reclamation } from '../models/Reclamation';  // Make sure to create this model

@Injectable({
  providedIn: 'root'
})
export class ReclamationService {

  private apiUrl = 'http://localhost:8081/api/reclamations'; // Adjust the API base URL

  constructor(private http: HttpClient) { }

  // Get all reclamations
  getAllReclamations(): Observable<Reclamation[]> {
    return this.http.get<Reclamation[]>(this.apiUrl);
  }

  // Get a single reclamation by ID
  getReclamationById(id: number): Observable<Reclamation> {
    return this.http.get<Reclamation>(`${this.apiUrl}/${id}`);
  }

  // Create a new reclamation
  createReclamation(reclamation: Reclamation): Observable<Reclamation> {
    return this.http.post<Reclamation>(this.apiUrl, reclamation);
  }

  // Update an existing reclamation by ID
  updateReclamation(id: number, updatedReclamation: Reclamation): Observable<Reclamation> {
    return this.http.put<Reclamation>(`${this.apiUrl}/${id}`, updatedReclamation);
  }

  // Delete a reclamation by ID
  deleteReclamation(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
